[Docs] Add intro and quickstart docs (#859)
mui/mui-toolpad: docs/data/toolpad/connecting-to-data-sources/fetch.md

# Fetch

> <p class="description">
>     <b>Fetch</b> query type is an easiest and fastest way to load external data within Toolpad app
> </p>

Reviewer: A few changes: fix the missing leading dot, fix the formatting, remove it in the description, and fix a small grammar issue.

```suggestion
<p class="description">Fetch query type is the easiest and fastest way to load external data within Toolpad app.</p>
```

*x the other pages*

As explained in the [connections](/toolpad/connecting-to-data-sources/connections/) section you can either create a reusable `connection` or simply create a new `query` and put all connection details inline

1. Choose `ADD QUERY` in the `Properties editor` on the right

1. Select `Fetch` type and click `CREATE QUERY`

>    ![Fetch type](/static/toolpad/fetch-query-1.png)

Reviewer: For the screenshots, I don't think that we can use the `![]()`. I think that we need the images to be x2 the screen resolution to not look blurry (it looks blurry on my end). I think that this one is a good example https://github.com/mui/material-ui/blob/660b9da2485257c80ba8be5e1079756ccb07f85d/docs/data/base/guides/working-with-tailwind-css/working-with-tailwind-css.md?plain=1#L311

*x the other screenshots*

Retool doesn't do it very well, e.g. https://docs.retool.com/docs/analytics#configuration has the right pixel resolution, but the display dimension is wrong.

Author: <img width="864" alt="image" src="https://user-images.githubusercontent.com/437214/187864293-f601da61-655a-408f-b02b-1bff8d0d7dae.png">

Here is the result - might not be so easy to see from screenshot but I think it's visible - the second screenshot was captured on laptop, so it seems that it really depends what sort of resolution you use when capturing screenshot. I'm still using `![]()` for both screenshots, so it seems that this should not be a problem. Unless you mean that we should use a fixed dimensions so it doesn't grow too big and mess up pixels? 🤔

Reviewer: > my external monitor vs. laptop screen

@bytasv Considering the resolution of your external monitor on the screenshot (`window.devicePixelRatio = 1`), I think that you will need to take the screenshot on your laptop screen. The PPI of your external monitor seems too low: https://bjango.com/articles/macexternaldisplays2/. For example, the latest MacBook Pro is 253.93 PPI (`window.devicePixelRatio = 2`), but the [external monitors](https://www.sharpnecdisplays.eu/p/eeme/en/products/details/t/desktop-displays/enterprise-displays/rp/ea271u.xhtml) that Michal purchased are 163.18 PPI, still not hight enough (I assume `window.devicePixelRatio = 1.5`. 

> I'm wondering if the problem is with `![]()`

My bad, I was confusing. If we take high resolution screenshots (which I would argue we should) then we can't use `![]()`. Why? Because if the element is 100 CSS px wide, the screenshot will be 200 px wide, which once displayed is 200 CSS px wide, but it needs to be 100 CSS px wide.  So we need to use `<img src="" style="width: 100px" >`, e.g. https://github.com/mui/material-ui/blob/660b9da2485257c80ba8be5e1079756ccb07f85d/docs/data/base/guides/working-with-tailwind-css/working-with-tailwind-css.md?plain=1#L311. `![]()` only works if the screenshot width is x2 larger than the max width of the container.

Author: That makes sense (both points), I'm planning to do a follow-up on images separately from this PR

Reviewer 2: > @michaldudak, I'm curious, what do you have for window.devicePixelRatio for a console log inside one of your external monitors?).

It depends on the scaling I set in Windows. I personally use them at 1.5.

# Fetch

<p class="description">Fetch query type is the easiest and fastest way to load external data within Toolpad app.</p>

> ## Working with fetch

Reviewer 3: This section has become a bit outdated due to [recent changes](https://github.com/mui/mui-toolpad/pull/721) to the fetch datasource.

I'm fine with merging as is and revisiting this as follow-up

Author: Yeah, let's merge as is as I'm now afraid it can get even more out-dated. I can fix this up in a follow up
